syntax = "proto3";

import "header.proto";

message GoToRequest {
    Header header = 1;
    int32 x_coord = 2;
    int32 y_coord = 3;
}

message GoToResponse {
    enum GoToResult {
        SUCCESS = 0;
        INVALID_COORDINATES = 1;
        CANNOT_MOVE = 2;
    }
    Header header = 1;
    GoToResult gotoResult = 2;
}

service GoToController {
    rpc GoToCoordinates(GoToRequest) returns (GoToResponse) {}
}
